File: jserver-framework/src/main/java/com/github/gr3gdev/jserver/http/RemoteAddress.java
package com.github.gr3gdev.jserver.http;

/**
 * Interface for remote address.
 *
 * @author Gregory Tardivel
 */
public interface RemoteAddress {

    /**
     * Remote socket address IP.
     *
     * @return ip
     */
    String ip();

    /**
     * Remote socket address port.
     *
     * @return port
     */
    String port();

}

File: jserver-core/src/main/kotlin/com/github/gr3gdev/jserver/http/RemoteAddress.kt
package com.github.gr3gdev.jserver.http

/**
 * Interface for remote address.
 *
 * @author Gregory Tardivel
 */
interface RemoteAddress {

    /**
     * Remote socket address IP.
     *
     * @return ip
     */
    fun ip(): String

    /**
     * Remote socket address port.
     *
     * @return port
     */
    fun port(): String

}